test/IRGen/framepointer.sil

```diff
@@ -33,7 +33,9 @@ entry(%i : $Int32):
 // CHECK:   ret i32 %1
 // CHECK: }
 
-// CHECK-SYSV: attributes [[ATTR]] = { {{.*}}"frame-pointer"="all"
+// CHECK-SYSV-macosx: attributes [[ATTR]] = { {{.*}}"frame-pointer"="all"
+// CHECK-SYSV-linux-gnu: attributes [[ATTR]] = { {{.*}}"frame-pointer"="all"
+// CHECK-SYSV-linux-android: attributes [[ATTR]] = { {{.*}}"frame-pointer"="non-leaf"
```

Comment:
The Android default appears to be different; one of the few changes in this pull that affects other non-Android platforms too, in that each one will need to be listed here.

Reply:
Modified this so that it checks all the officially supported platforms, plus Android, but it's no longer required to list each platform.